packages/nodes-base/nodes/Mattermost/Mattermost.node.ts

```diff
@@ -1,2396 +1,26 @@
 import {
-	IExecuteFunctions,
-} from 'n8n-core';
-
-import {
-	IDataObject,
-	ILoadOptionsFunctions,
-	INodeExecutionData,
-	INodePropertyOptions,
-	INodeType,
-	INodeTypeDescription,
-	NodeOperationError,
+	INodeTypeBaseDescription,
+	INodeVersions,
 } from 'n8n-workflow';
 
-import {
-	apiRequest,
-	apiRequestAllItems,
-	IAttachment,
-} from './GenericFunctions';
-
-import {
-	snakeCase,
-} from 'change-case';
-
-export class Mattermost implements INodeType {
-	description: INodeTypeDescription = {
-		displayName: 'Mattermost',
-		name: 'mattermost',
-		icon: 'file:mattermost.svg',
-		group: ['output'],
-		version: 1,
-		subtitle: '={{$parameter["operation"] + ": " + $parameter["resource"]}}',
-		description: 'Sends data to Mattermost',
-		defaults: {
-			name: 'Mattermost',
-			color: '#000000',
+import { MattermostV1 } from './v1/MattermostV1.node';
+import { NodeVersionedType } from '../../src/NodeVersionedType';
+
+export class Mattermost extends NodeVersionedType {
+	constructor() {
+		const description: INodeTypeBaseDescription = {
+			displayName: 'Mattermost V1',
+			name: 'mattermost',
+			icon: 'file:mattermost.svg',
+			group: ['output'],
+			subtitle: '={{$parameter["operation"] + ": " + $parameter["resource"]}}',
+			description: 'Sends data to Mattermost',
+			defaultVersion: 1,
```

Comment:
Why do we need to specify the default version...? 
1. in the `description` at L18, 
2. in the assignment at L23, and 
3. as the third arg to `super()` at L24

I would refactor to have it only in the `description` and use it always from there - but maybe I am missing something.